### DISCUSSION TITLE
Websockets and Nest.js with Mikro-ORM

### DISCUSSION THREAD

**Discussion by sergeyampo** [mikro-orm/mikro-orm#3055] [Q&A]
Hi, is there any instructions with using websocket(via socket.io), nest.js and mikro-orm?
What advices can you give?
How should I implement request scope and identity map?

**Comment by bruce-shi** [comment]
According to this guide: https://docs.nestjs.com/recipes/mikroorm#using-asynclocalstorage-for-request-context.
You need inject the `orm` and decorate each handler with `UseRequestContext`

```typescript
@WebSocketGateway()
export class AppGateway {
  constructor(private orm: MikroORM) {}
  
  @WebSocketServer()
  server: Server;

  @SubscribeMessage('Event')
  @UseRequestContext()
  async onEvent(client: Socket, payload: string): WsResponse {
    return { event: 'Event', data: {} };
  }
}
```

**Reply by michaelgambold** [reply]
Can confirm this works. The only downside is vs code says "Property 'orm' is declared but its value is never read." in the constructor

**Reply by Sergionx** [reply]
Now on 2026, the decorator was renamed to `@CreateRequestContext()`
